api/controller/psmdb_cluster.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package percona.platform.dbaas.controller.v1beta1;
+
+option go_package = "controller;controllerv1beta1";
+
+import "controller/common.proto";
+import "github.com/mwitkow/go-proto-validators/validator.proto";
+
+// PSMDBClusterState represents PSMDB cluster CR state.
+enum PSMDBClusterState {
+  // PSMDB_CLUSTER_STATE_INVALID represents unknown state.
+  PSMDB_CLUSTER_STATE_INVALID = 0;
+  // PSMDB_CLUSTER_STATE_CHANGING represents a cluster being changed (initializing).
+  PSMDB_CLUSTER_STATE_CHANGING = 1;
+  // PSMDB_CLUSTER_STATE_READY represents a cluster without pending changes (ready).
+  PSMDB_CLUSTER_STATE_READY = 2;
+  // PSMDB_CLUSTER_STATE_FAILED represents a failed cluster (error).
+  PSMDB_CLUSTER_STATE_FAILED = 3;
+  // PSMDB_CLUSTER_STATE_DELETING represents a cluster being deleting.
+  PSMDB_CLUSTER_STATE_DELETING = 4;
+}
+
+// PSMDBBackupState represents PSMDB backup CR state.
+enum PSMDBBackupState {
+  // PSMDB_BACKUP_STATE_INVALID represents unknown state.
+  PSMDB_BACKUP_STATE_INVALID = 0;
+  // PSMDB_BACKUP_STATE_RUNNING represents running backup (Starting, Running).
+  PSMDB_BACKUP_STATE_RUNNING = 1;
+  // PSMDB_BACKUP_STATE_SUCCEEDED represents succeeded backup (Succeeded).
+  PSMDB_BACKUP_STATE_SUCCEEDED = 2;
+  // PSMDB_BACKUP_STATE_FAILED represents failed backup (Failed).
+  PSMDB_BACKUP_STATE_FAILED = 3;
+}
+
+// PSMDBClusterParams represents PSMDB cluster parameters that can be updated.
+message PSMDBClusterParams {
+  // ReplicaSet container parameters.
+  message ReplicaSet {
+    // Requested compute resources.
+    ComputeResources compute_resources = 1 [
+      (validator.field) = {
+        msg_exists: true
```

Comment:
This might be optional in case of minikube? 

https://www.percona.com/doc/kubernetes-operator-for-psmongodb/minikube.html
> comment resources.requests.memory and resources.requests.cpu keys (this will fit the Operator in minikube default limitations)

https://www.percona.com/doc/kubernetes-operator-for-psmongodb/api.html#create-new-psmdb-cluster
> "resources": {
               "limits": null
            },